backend/src/test/java/com/google/models/AnomalyTest.java

```diff
@@ -135,6 +136,17 @@ public void createAnomalyFromEmbeddedEntity_correctEmbeddedEntityToAnomalyConver
     assertEquals(convertedAnomaly, ANOMALY);
   }
 
+  @Test
+  public void createAnomalyFromEmbeddedEntity_dataPointsInChronologicalOrder() {
+    EmbeddedEntity anomalyEmbeddedEntity = ANOMALY.toEmbeddedEntity();
+    Anomaly convertedAnomaly = Anomaly.createAnomalyFromEmbeddedEntity(anomalyEmbeddedEntity);
+
+    assertEquals(
+      convertedAnomaly.getDataPoints().keySet().stream().sorted().collect(Collectors.toList()), 
```

Comment:
Maybe this should be the expected data instead of sorting the anomaly's data points and checking that with the actual data points? I'm assuming `sorted()` will use the Timestamp comparator to sort, which is what the TreeMap also uses, so my impression of this test is that it's not really proper / sound since it's kind of just doing the same thing. But I am probably wrong about this so feel free to comment / explain.

Reply:
okay, i think what you said makes sense. I will compare the sorted list to a list of datapoints that is manually sorted, i.e. in the case of the tests, just the order of `DUMMY_DATA_POINTS` which is in the correct order.

thanks!